site/source/docs/contributing/developers_guide.rst

```diff
@@ -95,6 +98,46 @@ automatically when you create a pull request, and they should all pass. If you
 run into trouble with a test failure you can't fix, please let the developers
 know.
 
+Bisecting
+=========
+
+If you find a regression, bisection is often the fastest way to figure out what
+went wrong. This is true not just for finding an actual regression in Emscripten
+that you want to fix, but also if your project stopped working when you upgrade,
+and you need to investigate if it's an Emscripten regression or something else.
+
+The normal ``git bisect`` workflow can work if you just need to bisect a single
+repository. For example, you can bisect only on the emscripten repo if you are on
+a range that all works with the same versions of LLVM and Binaryen (which was
+mentioned earlier, are two separate codebases that are depended on).
+
+If you have a large bisection range, you generally can't bisect a single repo.
+You can still bisect, though! To do that you need the emsdk and to understand
+how the
+`release process <https://github.com/emscripten-core/emscripten/blob/master/docs/process.md#release-processes>`_
+works for all the repos together. The key "trick" is that
+
+::
+
+     emsdk install tot
+
+can install an arbitrary build of emscripten: it installs the one identified
+in ``emscripten-releases-tot.txt``. You can therefore bisect on the
+`releases repo <https://chromium.googlesource.com/emscripten-releases>`_ which
+has a DEPS file that basically says what version of all the various repos are
+in which release. In each bisection step on that repo you get the git hash in
+that repo, which is the identifier of the corresponding ``tot`` build (which
+when it was built, was tip-of-tree).
+
+Each step in this bisection will download a complete build which is not a
+trivial download. However, at least the number of such steps will be
+logarithmic!
+
+This bisects down to a single commit in the releases repo. That commit will
```

Comment:
I was expecting an actual script or recipe for how run this bisect?   Do you do it by hand?   Maybe we could give an example of the commands to run?

Reply:
I'll add a mention and link to the `git bisect` docs. Is it worth adding more? I think it's better to leave that to the official docs.

Reply:
No I meant like an example of what you actually run... I imagine its fairly complex, no?  Like maybe an example?  I was imaging something like this:

```
<emscripten-release> $ release=$(git rev-parse HEAD)
<emscripten-release> $ ../emsdk/emsdk install upstream-$release
<emscripten-release> $ ../emsdk/emsdk activate upstream-$release
<emscripten-release> $ ../my_project/run_my_test.sh  # run test using active emsdk
<emscripten-release> $ git bisect bad
<emscripten-release> $ ... etc
```

Is that about right?  If yes, maybe its obvious.. if I misunderstood maybe we need more clarity.

Reply:
Yeah, basically that, but I'm not sure what `git show DEPS` does, and FWIW my personal process is somewhat different (involving multiple console tabs etc.). Overall I don't think writing out concrete steps would help here, and I'm unsure how concrete to be, so I focused on the ideas.

But I can just paste what you wrote, if you think it'll help?

Reply:
`git show` shows the previous change.. `git show DEPS` shows the previous change but limits it to just showing the `DEPS` file.  

I think its probably fine... its tricky to write a script across multiple repos which is what this requires and I think your explanation makes sense.. at least I understood it :)